internal/report/store.go

```diff
@@ -89,23 +89,62 @@ func (s *PolicyReportStore) AddClusterPolicyReport(report *ClusterPolicyReport)
 	return nil
 }
 
+func (s *PolicyReportStore) lookForNamespacedReport(namespace string) (PolicyReport, bool) {
+	s.namespacedPolicyReportsMutex.Lock()
+	defer s.namespacedPolicyReportsMutex.Unlock()
+	if report, ok := s.namespacedPolicyReports[namespace]; ok {
+		return report, true
+	}
+	return PolicyReport{}, false
+}
+
 // Get PolicyReport by namespace
 func (s *PolicyReportStore) GetPolicyReport(namespace string) (PolicyReport, error) {
-	s.namespacedPolicyReportsMutex.RLock()
-	defer s.namespacedPolicyReportsMutex.RUnlock()
-	report, present := s.namespacedPolicyReports[namespace]
-	if present {
+	// check if we already have a report to the given namespace in the cache
+	if report, ok := s.lookForNamespacedReport(namespace); ok {
 		return report, nil
 	}
-	return PolicyReport{}, errors.New("report not found")
+	// nothing in the cache. Let's request it in the cluster
+	report := polReport.PolicyReport{}
+	getErr := s.client.Get(context.TODO(), types.NamespacedName{
+		Namespace: namespace,
+		Name:      getNamespacedReportName(namespace),
+	}, &report)
+	if getErr != nil {
+		if errorMachinery.IsNotFound(getErr) {
+			return PolicyReport{}, errors.New("report not found")
+		}
+		return PolicyReport{}, getErr
+	}
+	policyReport := &PolicyReport{
+		PolicyReport: report,
+	}
+	if err := s.AddPolicyReport(policyReport); err != nil {
+		return PolicyReport{}, errors.New("fetcher report cannot store report on cache")
+	}
+
+	return *policyReport, nil
 }
 
 // Get the ClusterPolicyReport
 func (s *PolicyReportStore) GetClusterPolicyReport() (ClusterPolicyReport, error) {
 	s.clusterPolicyReportMutex.RLock()
 	defer s.clusterPolicyReportMutex.RUnlock()
-	report := s.clusterPolicyReport
-	return report, nil
+	fmt.Print(s.clusterPolicyReport.ClusterPolicyReport.ObjectMeta.Name)
+	if len(s.clusterPolicyReport.ClusterPolicyReport.Results) == 0 {
```

Comment:
Also in this case, I would just use the cache provided by the kubernetes' go client